tbd_tracker_node:
  ros__parameters:
    tracker:
      frame_id: 'map'
      mismatch_penalty: 100.
      assignment_algo: 'hungarian'
      yaw_corr: 'False'
      publishers:
        names: ['track_pub']
        track_pub:
          pub_topic: 'tracks'
          msg_type: 'Tracks3D'
          function: 'publish_tracks'
          queue_size: 10

    object_properties:
      object_classes: ['bicycle','bus','car','motorcycle','pedestrian','trailer','truck']

      bicycle:
        model_type: 'cvcy'
        yaw_proc_var: [1.]
        size_proc_var: [1., 1., 1.]
        vel_proc_var: [1., 1., 1.]
        create_method: 'conf'
        delete_method: 'conf'
        active_thresh: .75
        delete_thresh: .15
        detect_thresh: .15
        score_update_function: 'parallel_add'
        score_decay: .05
        sim_metric: 'dist_3d'
        match_thresh: 5.0

      bus:
        model_type: 'cvcy'
        yaw_proc_var: [1.]
        size_proc_var: [1., 1., 1.]
        vel_proc_var: [1., 1., 1.]
        create_method: 'conf'
        delete_method: 'conf'
        active_thresh: .75
        delete_thresh: .15
        detect_thresh: .15
        score_update_function: 'parallel_add'
        score_decay: .05
        sim_metric: 'dist_3d'
        match_thresh: 5.0

      car:
        model_type: 'cvcy'
        yaw_proc_var: [1.]
        size_proc_var: [1., 1., 1.]
        vel_proc_var: [1., 1., 1.]
        create_method: 'conf'
        delete_method: 'conf'
        active_thresh: .75
        delete_thresh: .15
        detect_thresh: .15
        score_update_function: 'parallel_add'
        score_decay: .05
        sim_metric: 'dist_3d'
        match_thresh: 5.0

      motorcycle:
        model_type: 'cvcy'
        yaw_proc_var: [1.]
        size_proc_var: [1., 1., 1.]
        vel_proc_var: [1., 1., 1.]
        create_method: 'conf'
        delete_method: 'conf'
        active_thresh: .75
        delete_thresh: .15
        detect_thresh: .15
        score_update_function: 'parallel_add'
        score_decay: .05
        sim_metric: 'dist_3d'
        match_thresh: 5.0

      pedestrian:
        model_type: 'cvcy'
        yaw_proc_var: [1.]
        size_proc_var: [1., 1., 1.]
        vel_proc_var: [1., 1., 1.]
        create_method: 'conf'
        delete_method: 'conf'
        active_thresh: .75
        delete_thresh: .15
        detect_thresh: .15
        score_update_function: 'parallel_add'
        score_decay: .05
        sim_metric: 'dist_3d'
        match_thresh: 5.0

      trailer:
        model_type: 'cvcy'
        yaw_proc_var: [1.]
        size_proc_var: [1., 1., 1.]
        vel_proc_var: [1., 1., 1.]
        create_method: 'conf'
        delete_method: 'conf'
        active_thresh: .75
        delete_thresh: .15
        detect_thresh: .15
        score_update_function: 'parallel_add'
        score_decay: .05
        sim_metric: 'dist_3d'
        match_thresh: 5.0

      truck:
        model_type: 'cvcy'
        yaw_proc_var: [1.]
        size_proc_var: [1., 1., 1.]
        vel_proc_var: [1., 1., 1.]
        create_method: 'conf'
        delete_method: 'conf'
        active_thresh: .75
        delete_thresh: .15
        detect_thresh: .15
        score_update_function: 'parallel_add'
        score_decay: .05
        sim_metric: 'dist_3d'
        match_thresh: 5.0

    detectors:
      detector_names: ['megvii']
      megvii:
        topic: 'converted_detections'
        msg_type: 'Detections3D'
        detector_type: 'pos_bbox_3d'
        detection_classes: ['barrier','bicycle','bus','car','construction_vehicle','motorcycle','pedestrian','traffic_cone','trailer','truck']
        detection_properties:
          barrier:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: True
            object_class: 
          bicycle:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: False
            object_class: bicycle
          bus:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: False
            object_class: bus
          car:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: False
            object_class: car
          construction_vehicle:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: True
            object_class: 
          motorcycle:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: False
            object_class: motorcycle
          pedestrian:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: False
            object_class: pedestrian
          traffic_cone:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: True
            object_class: 
          trailer:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: False
            object_class: trailer
          truck:
            pos_obs_var: [1., 1., 1.]
            yaw_obs_var: [1.]
            size_obs_var: [1., 1., 1.]
            ignore: False
            object_class: truck
